clippy_lints/src/match_on_vec_items.rs

```diff
@@ -87,3 +86,15 @@ fn is_vec_indexing<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr<'tcx>)
 
     None
 }
+
+fn is_vector(cx: &LateContext<'_, '_>, expr: &Expr<'_>) -> bool {
+    let ty = cx.tables.expr_ty(expr);
+    let ty = walk_ptrs_ty(ty);
+    is_type_diagnostic_item(cx, ty, sym!(vec_type))
+}
+
+fn is_full_range(cx: &LateContext<'_, '_>, expr: &Expr<'_>) -> bool {
+    let ty = cx.tables.expr_ty(expr);
+    let ty = walk_ptrs_ty(ty);
+    match_type(cx, ty, &["core", "ops", "range", "RangeFull"])
```

Comment:
Does this work as well?

```suggestion
    match_type(cx, ty, &utils::paths::RANGE_FULL)
```

Reply:
No, this doesn't work
```Rust
pub const RANGE_FULL: [&str; 3] = ["core", "ops", "RangeFull"];
```

Reply:
Then the path in `util::paths` is wrong. Can you change that and use it here afterwards?

Reply:
Updated it